src/types.md

```diff
@@ -236,3 +236,311 @@ fn main() {
 
 Of course, enumerating all possible stored variants remains preferable such that the
 compiler helps you to avoid runtime panics.
+
+## When should I put my data in a `Box`?
+
+In C++, you often need to box things for ownership reasons, whereas in Rust
+it's typically just a performance trade-off. It's arguably premature optimization
+to use boxes unless your profiling shows a lot of memcpy of that particular
+type (or, perhaps, the relevant [clippy lint](https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#large_enum_variant)
+informs you that you have a problem.)
+
+> I never box things unless it’s really big - MG
+
+Another heuristic is if part of your data structure is very rarely filled,
+in which case you may wish to `Box` it to avoid incurring an overhead for all
+other instances of the type.
+
+```rust
+# struct Humility; struct Talent; struct Ego;
+struct Popstar {
+  ego: Ego,
+  talent: Talent,
+  humility: Option<Box<Humility>>,
+}
+# fn main() {}
+```
+
+(This is one reason why people like using [anyhow](https://docs.rs/anyhow/latest/anyhow/)
+for their errors; it means the failure case in their `Result` enum is only
+a pointer wide.)
+
+Of course, Rust may require you to use a box:
+
+* if you need to `Pin` some data, typically for async Rust
+* if you otherwise have an infinitely sized data structure.
+
+but as usual, the compiler will explain very nicely.
+
+## I miss operator overloading! What do I do?
+
+Implement the standard traits instead. This has equivalent effect in that
```

Comment:
Give some examples of which standard traits you mean, e.g. `PartialEq`, `Add`, et c.

Reply:
Also, this could/should go right under the function overloading question. Or you could fold the 2 together.